src/snapshot/set.rs

```diff
@@ -67,7 +74,32 @@ impl Plugin for SnapshotSetPlugin {
             SaveWorld,
             (SaveWorldSet::Checksum, SaveWorldSet::Snapshot).chain(),
         )
+        .configure_sets(
+            AdvanceWorld,
+            (
+                AdvanceWorldSet::First,
+                AdvanceWorldSet::Main,
+                AdvanceWorldSet::Last,
+            )
+                .chain(),
+        )
         .add_systems(LoadWorld, apply_deferred.in_set(LoadWorldSet::EntityFlush))
-        .add_systems(LoadWorld, apply_deferred.in_set(LoadWorldSet::DataFlush));
+        .add_systems(LoadWorld, apply_deferred.in_set(LoadWorldSet::DataFlush))
+        .add_systems(
+            AdvanceWorld,
+            apply_deferred
+                .after(AdvanceWorldSet::First)
+                .before(AdvanceWorldSet::Main),
+        )
+        .add_systems(
+            AdvanceWorld,
+            apply_deferred
+                .after(AdvanceWorldSet::Main)
+                .before(AdvanceWorldSet::Last),
+        )
```

Comment:
Honest question: What's the practical difference between using `Schedule`s and sets, when they have hard sync points like apply_deferred between them?

Reply:
The differences (as I understand it):

- Sets must be run completely, and must be contained within a schedule. It is not possible to run all systems within `MySet::A` and not `MySet::B`.
- Systems can be ordered `before` and `after` sets, but this cannot be done for schedules.

So if we had `GgrsSchedule` on its own, there wouldn't be a simple way to schedule systems to execute `before` and `after` it runs. Adding a set, `AdvanceWorldSet` allows for controlled ordering, but then every system would need to be added in a explicit set if `AdvanceWorldSet` was inside `GgrsSchedule`. Conversely, if `GgrsSchedule` was within the `AdvanceWorldSet`, we'd have no way to trigger the schedule on demand. So we require a `Schedule` -> `Set` -> `Schedule` architecture to allow for `before` and `after` ordering, _and_ the ability to trigger the schedule whenever required.